flaml/tune/space.py

```diff
@@ -225,15 +225,18 @@ def add_cost_to_space(space: Dict, low_cost_point: Dict, choice_cost: Dict):
                 domain.choice_cost = cost[ind]
                 domain.const = [domain.const[i] for i in ind]
                 domain.ordered = True
-            elif all(
-                isinstance(x, int) or isinstance(x, float) for x in domain.categories
-            ):
-                # sort the choices by value
-                ind = np.argsort(domain.categories)
-                domain.categories = [domain.categories[i] for i in ind]
-                domain.ordered = True
             else:
-                domain.ordered = False
+                ordered = getattr(domain, "ordered", None)
+                if ordered is None:
+                    # automatically decide whether to order the choices based on the value type
+                    domain.ordered = ordered = all(
+                        isinstance(x, (int, float)) for x in domain.categories
+                    )
+                if ordered:

```

Comment:
```suggestion
                if ordered is True:

```

Reply:
add is True to tell apart from line 230

Reply:
What do you mean by "tell apart from line 230"?

Reply:
line 230 is "if ordered is None". If what you mean here is "if ordered is True" instead of "if ordered is not None", could make it more clear.

Reply:
`ordered` is bool or None. `if ordered` means `if ordered is True`. There is no ambiguity here.
In contrast, if we want to check whether `ordered` is False, we need to explicitly write `ordered is False` instead of `not ordered` because the latter indicates `ordered` is either False or None.
These two situations are different.